// client/src/components/LinkList.tsx
import { useEffect, useState } from 'react';
import {
  Box,
  Button,
  Card,
  CardBody,
  CardHeader,
  Heading,
  Stack,
  Text,
  useToast,
} from '@chakra-ui/react';
import { Link, linksApi } from '../services/api';

export function LinkList() {
  const [links, setLinks] = useState<Link[]>([]);
  const [isLoading, setIsLoading] = useState(true);
  const toast = useToast();

  const fetchLinks = async () => {
    try {
      const response = await linksApi.getAll();
      setLinks(response.data);
    } catch (error) {
      toast({
        title: 'Error fetching links',
        description: 'Please try again later',
        status: 'error',
        duration: 3000,
        isClosable: true,
      });
    } finally {
      setIsLoading(false);
    }
  };

  useEffect(() => {
    fetchLinks();
  }, []);

  const handleDelete = async (id: string) => {
    try {
      await linksApi.delete(id);
      toast({
        title: 'Link deleted successfully',
        status: 'success',
        duration: 3000,
        isClosable: true,
      });
      fetchLinks();
    } catch (error) {
      toast({
        title: 'Error deleting link',
        description: 'Please try again later',
        status: 'error',
        duration: 3000,
        isClosable: true,
      });
    }
  };

  if (isLoading) {
    return <Text>Loading...</Text>;
  }

  if (links.length === 0) {
    return <Text>No links found. Add your first link!</Text>;
  }

  return (
    <Stack spacing={4} w="100%" maxW="800px" mx="auto">
      {links.map((link) => (
        <Card key={link.id}>
          <CardHeader>
            <Heading size="md">
              <a href={link.url} target="_blank" rel="noopener noreferrer">
                {link.title}
              </a>
            </Heading>
          </CardHeader>
          <CardBody>
            {link.description && <Text mb={4}>{link.description}</Text>}
            <Button
              colorScheme="red"
              size="sm"
              onClick={() => handleDelete(link.id)}
            >
              Delete
            </Button>
          </CardBody>
        </Card>
      ))}
    </Stack>
  );
}
