// client/src/components/LinkForm.tsx
import { useState } from 'react';
import {
  Box,
  Button,
  FormControl,
  FormLabel,
  Input,
  Textarea,
  VStack,
  useToast,
} from '@chakra-ui/react';
import { linksApi, CreateLinkData } from '../services/api';

interface LinkFormProps {
  onSuccess: () => void;
}

export function LinkForm({ onSuccess }: LinkFormProps) {
  const [isLoading, setIsLoading] = useState(false);
  const toast = useToast();
  const [formData, setFormData] = useState<CreateLinkData>({
    url: '',
    title: '',
    description: '',
  });

  const handleSubmit = async (e: React.FormEvent) => {
    e.preventDefault();
    setIsLoading(true);

    try {
      await linksApi.create(formData);
      toast({
        title: 'Link created successfully',
        status: 'success',
        duration: 3000,
        isClosable: true,
      });
      setFormData({ url: '', title: '', description: '' });
      onSuccess();
    } catch (error) {
      toast({
        title: 'Error creating link',
        description: 'Please try again later',
        status: 'error',
        duration: 3000,
        isClosable: true,
      });
    } finally {
      setIsLoading(false);
    }
  };

  const handleChange = (
    e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>
  ) => {
    const { name, value } = e.target;
    setFormData((prev) => ({ ...prev, [name]: value }));
  };

  return (
    <Box as="form" onSubmit={handleSubmit} w="100%" maxW="600px" mx="auto">
      <VStack spacing={4}>
        <FormControl isRequired>
          <FormLabel>URL</FormLabel>
          <Input
            name="url"
            value={formData.url}
            onChange={handleChange}
            placeholder="https://example.com"
            type="url"
          />
        </FormControl>

        <FormControl isRequired>
          <FormLabel>Title</FormLabel>
          <Input
            name="title"
            value={formData.title}
            onChange={handleChange}
            placeholder="Enter title"
          />
        </FormControl>

        <FormControl>
          <FormLabel>Description</FormLabel>
          <Textarea
            name="description"
            value={formData.description}
            onChange={handleChange}
            placeholder="Enter description (optional)"
          />
        </FormControl>

        <Button
          type="submit"
          colorScheme="blue"
          isLoading={isLoading}
          w="100%"
        >
          Add Link
        </Button>
      </VStack>
    </Box>
  );
}
